src/Microsoft.Extensions.Configuration.Abstractions/ConfigurationExtensions.cs

```diff
@@ -55,5 +56,13 @@ public static string GetConnectionString(this IConfiguration configuration, stri
                 }
             }
         }
+
+        /// <summary>
+        /// Determines whether the section has a <see cref="IConfigurationSection.Value"/> or has children 
+        /// </summary>
+        public static bool Exists(this IConfigurationSection section)
+        {
+            return section !=null && section.Value != null || section.GetChildren().Any();
```

Comment:
Oops, will fix. What is the convention for squashing into one change? Keep all the commit messages or rewrite with a more appropriate one? And just force push to the same branch?